src/bridge/cockpitpackages.c

```diff
@@ -338,30 +338,16 @@ package_walk_directory (GChecksum *own_checksum,
 }
 
 static JsonObject *
-read_json_file (const gchar *directory,
-                const gchar *name,
+read_json_file (const gchar *path,
                 GError **error)
 {
-  JsonObject *object;
-  GMappedFile *mapped;
-  gchar *filename;
-  GBytes *bytes;
-
-  filename = g_build_filename (directory, name, NULL);
-  mapped = g_mapped_file_new (filename, FALSE, error);
-  g_free (filename);
+  g_autoptr(GMappedFile) mapped = g_mapped_file_new (path, FALSE, error);
```

Comment:
"This temporarily makes the code more complicated" +9 -22 lol.

Reply:
But that's due to g_auto() -- I meant the duplication of g_build_filename() . But I figure that was just a joke anyway :grin: